### ISSUE TITLE
request.queue_0.queue does not return results

### ISSUE THREAD

**Issue by sdrjason** [savonet/liquidsoap#2088]
Installed liquidsoap 2.0.0 via opam on Raspberry Pi OS Bullseye

I start liquidsoap with a socket interface (script below).  I can add files to the queue and they play correctly, but the request.queue_0.queue command doesn't return the list of files in the queue (the command appears to complete without error but it never returns a list).  I get the same results using the socket or telnet interface.  

To reproduce, here is the script I use to start liquidsoap:
```
#!/home/pi/.opam/4.10.0/bin/liquidsoap

# Example liquidsoap streaming from op25 to icecast
# (c) 2019-2021 gnorbury@bondcar.com, wllmbecks@gmail.com
#

log.level.set(4)
log.file.set(false)
log.stdout.set(true)

# create a socket to send commands to this instance of liquidsoap
settings.server.socket.set(true)
settings.server.socket.path.set("/home/pi/socket")

# This creates a 1 second silence period generated programmatically (no disk reads) 
silence = blank(duration=1.)

# create the queue
myqueue = request.queue()

# If there is anything in the queue, play it.  If not, play the silence defined above repeatedly:
stream = fallback(track_sensitive=false, [myqueue, silence])

stream = compand(stream, mu = 0.5)

# ICECAST STREAMING
output.icecast(%mp3(bitrate=16, samplerate=22050, stereo=false), description="op25", genre="Public Safety", url="", fallible=false, icy_metadata="false", host="localhost", port=8000, mount="op25", password="op25source", stream)
```
Here is how I add files or check the queue:
```
#!/usr/bin/python
import socket
import sys
import os
import re

filepath = sys.argv[1]
filename = os.path.basename(filepath)
a = filename.split('-')
tg,junk = a[0],a[1]

# Create a UDS socket
sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)

# Connect the socket to the port where the server is listening
server_address = '/home/pi/socket'
print >>sys.stderr, 'connecting to %s' % server_address
try:
    sock.connect(server_address)
except socket.error, msg:
    print >>sys.stderr, msg
    sys.exit(1)

try:
    
    # Send data
    message = 'request.queue_0.push {0}\n'.format(filepath)
    # uncomment below and comment above to check the queue
    #message = 'request.queue_0.queue \n'
    print >>sys.stderr, 'sending "%s"' % message
    sock.sendall(message)

    amount_received = 0

    while not re.search('END', data):
        data = sock.recv(16)
        amount_received += len(data)
        print >>sys.stderr, 'received "%s"' % data    

finally:
    print >>sys.stderr, 'closing socket'
    sock.close()
```
Here is the telnet session (I get the same results on the socket interface, but the telnet format is easier to cut and paste).  There were files queued and playing when I ran this:
```
pi@raspberrypi:~ $ telnet localhost 1234
Trying ::1...
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
help
Available commands:
| exit
| help [<command>]
| list
| op25.metadata
| op25.remaining
| op25.skip
| quit
| request.alive
| request.all
| request.metadata <rid>
| request.on_air
| request.queue_0.push <uri>
| request.queue_0.queue
| request.queue_0.skip
| request.resolving
| request.trace <rid>
| uptime
| var.get
| var.list
| var.set <name> = <value>
| version

Type "help <command>" for more information.
END
request.queue_0.queue

END
quit
Bye!
Connection closed by foreign host.
```

Any help would be appreciated.

**Comment by toots** [comment]
Thanks for this report. It is most likely an issue with the default telnet interfaces registering several telnet operators using the same name with the last one hiding the previous ones.

I will have a look at this as part of the `2.0.2` release cycle. Meanwhile, you can try to mess around the `telnet.liq` file where all these operators are registered: https://github.com/savonet/liquidsoap/blob/v2.0.1/libs/telnet.liq

**Comment by toots** [comment]
Thanks for this report. I'll have a look at it as part of the `2.0.2` release cycle. This should be easy to fix.

**Comment by toots** [comment]
Thanks for the report. This issue is now fixed and will be release with the upcoming `2.0.2` release.

Shall you need to fix it meanwhile, you can insert this code at the beginning of your script:
```ruby
# Play a queue of uris. Returns a function to push new uris in the queue as well as the resulting source.
# @category Source / Track Processing
# @param ~id Force the value of the source ID.
# @param ~interactive Should the queue be controllable via telnet?
# @param ~prefetch How many requests should be queued in advance.
# @param ~native Use native implementation.
# @param ~queue Initial queue of requests.
# @param ~timeout Timeout (in sec.) for a single download.
# @method push Push a request on the request queue.
# @method length Length of the queue.
def request.queue(~id=null(), ~interactive=true, ~prefetch=1, ~native=false, ~queue=[], ~timeout=20.)
  id = string.id.default(default="request.queue", id)
  queue = ref(queue)
  fetch = ref(fun () -> true)
  def next() =
    if !queue != [] then
      let [r, ...q] = !queue
      queue := q
      (r:request)
    else
      null()
    end
  end
  def push(r)
    log.info(label=id, "Pushing #{r} on the queue.")
    queue := [...!queue, r]
    fn = !fetch
    ignore(fn())
  end
  def push_uri(uri)
    r = request.create(uri)
    push(r)
  end
  s =
    if native then
      stdlib_native.request.dynamic(id=id, timeout=timeout, next)
    else
      request.dynamic(id=id, prefetch=prefetch, timeout=timeout, available={not list.is_empty(!queue)}, next)
    end
  source.set_name(s, "request.queue")
  fetch := s.fetch
  if interactive then
    namespace = if id == "" then "queue" else id end
    def push(uri)
      r = request.create(uri)
      push(r)
      "#{request.id(r)}"
    end
    s.on_get_ready({
      begin
        server.register(namespace=s.id(), description="Flush the queue and skip the current track",
                        "flush_and_skip", fun (_) -> try
                          s.set_queue([])
                          s.skip()
                          "Done."
                        catch err do
                          "Error while flushing and skipping source: #{err}"
                        end)
        server.register(namespace=namespace, description="Push a new request in the queue.", usage="push <uri>", "push", push)
        def show_queue(_) =
          queue = [...s.queue(), ...(!queue)]
          string.concat(separator=" ", list.map(fun (r) -> string_of(request.id(r)), queue))
        end
        server.register(namespace=namespace, description="Display current queue content.", usage="queue", "queue", show_queue)
        def skip(_) = s.skip(); "Done." end
        server.register(namespace=namespace, description="Skip current song.", usage="skip", "skip", skip)
      end
    })
  end
  s.{push=push.{uri=push_uri}, length={list.length(!queue)}}
end
```

**Comment by hayden-t** [comment]
On a somewhat unrelated point @sdrjason i enjoyed reading over your example but i found a problem with runnnig

```
    while not re.search('END', data):
        data = sock.recv(16)
        amount_received += len(data)
        print >>sys.stderr, 'received "%s"' % data    
```

data is not defined 

```
trunk_recorder    | Traceback (most recent call last):
trunk_recorder    |   File "./streamthis.py", line 70, in <module>
trunk_recorder    |     while not re.search('END', data):
trunk_recorder    | NameError: name 'data' is not defined
```

instead i changed it to

```
    while True:
        data = sock.recv(16)
        if (re.search('END', data.decode())):
           break
        print(data.decode())
```
